src/mount_efs/__init__.py

```diff
@@ -599,12 +661,13 @@ def check_unsupported_options(options):
 
 
 def main():
-    fs_id, path, mountpoint, options = parse_arguments()
     assert_root()
 
     config = read_config()
     bootstrap_logging(config)
 
+    fs_id, path, mountpoint, options = parse_arguments(config)
```

Comment:
No, the reason I moved it is because `parse_arguments` now requires to get the `config` passed into it. Furthermore, it requires the logging to be setup due to the use of `fatal_error`.

This meant that I needed to move

```
    config = read_config()
    bootstrap_logging(config)
```

up above `parse_arguments`. This in turn meant that calling `mount.efs` as non-root would try to setup the logging, and then fail as I had described [in my previous comment](https://github.com/aws/efs-utils/pull/14#issuecomment-414252207).

Moving `assert_root` was an easy way to give the user feedback, without printing an ugly stacktrace due to missing permissions.

Please let me know if you have any other ideas on how this could be solved, which wouldn't potentially require a big refactoring.